### PR TITLE
Add MetadataURL to account struct

### LINKED ISSUE #1038: Users SBAT link to metadata
### Expected Behavior

Users should be able to communicate various information to other users in the ecosystem as specified in https://github.com/celo-org/CIPs/pull/1 and https://github.com/celo-org/CIPs/pull/4:
- The service URL of the attestation service for a validator
- The name/affiliation/picture of the validator, etc.

As per https://github.com/celo-org/CIPs/pull/4, we should:
- Add the ability to set a `metadataURL` property on the `Account` struct
- Have the metadata contain the above information.

### Current Behavior

There is no way to do this.

### PR DESCRIPTION
**Pull request by nambrot**
### Description

This PR adds the ability for accounts to register a URL under which relevant metadata can live

### Tested

- Unit tests

### Other changes

Added `setAccountDataEncryptionKey` in the interface.

### Related issues

- Fixes #1038

### CODE REVIEW

File: packages/protocol/contracts/identity/Attestations.sol [hunk 1]
```diff
@@ -87,6 +92,9 @@ contract Attestations is IAttestations, Ownable, Initializable, UsingRegistry, R
 
     // The ECDSA public key used to encrypt and decrypt data for this account
     bytes dataEncryptionKey;
+
+    // The URL under which an account add metadata and claims
```
Comment (m-chrzan) [inline packages/protocol/contracts/identity/Attestations.sol]: Typo: add*s*

File: packages/protocol/contracts/identity/interfaces/IAttestations.sol [hunk 2]
```diff
@@ -16,12 +16,13 @@ interface IAttestations {
 
   // TODO: For some reason the compiler will complain about the next line
   // function setAccountDataEncryptionKey(bytes) external;
+  // function setMetadataURL(string calldata) external;
```
Comment (m-chrzan) [inline packages/protocol/contracts/identity/interfaces/IAttestations.sol]: Why is this commented out?

File: packages/protocol/contracts/identity/Attestations.sol [hunk 3]
```diff
@@ -481,6 +489,24 @@ contract Attestations is IAttestations, Ownable, Initializable, UsingRegistry, R
     emit AttestationExpirySecondsSet(_attestationExpirySeconds);
   }
 
+  /**
+   * @notice Setter for the metadata of an account.
+   * @param metadataURL The URL to access the metadata.
+   */
+  function setMetadataURL(string memory metadataURL) public {
```
Comment (m-chrzan) [inline packages/protocol/contracts/identity/Attestations.sol]: Doesn't look like this is being used anywhere else in the contract. Is this a requirement for every account and so should be set when creating one? If not, function should be `external`.

File: packages/protocol/contracts/identity/interfaces/IAttestations.sol (hunk 2 above)
Comment (nambrot) [reply packages/protocol/contracts/identity/interfaces/IAttestations.sol]: The compiler complained about non matching byte codee, same with setting the encryption key ...

File: packages/protocol/contracts/identity/Attestations.sol (hunk 3 above)
Comment (nambrot) [reply packages/protocol/contracts/identity/Attestations.sol]: Not every account, but external makes sense!

**Review by m-chrzan — APPROVED** [review]

**Pull request merged**